WcaOnRails/app/webpacker/javascript/edit-events/modals/index.js

```diff
@@ -20,11 +20,9 @@ let RoundAttributeComponents = {
   advancementCondition: AdvancementConditionComponents,
 };
 
-function findRoundsSharingTimeLimitWithRound(wcifEvents, wcifRound) {
-  return _.compact(_.flatMap(wcifEvents, 'rounds')).filter(otherWcifRound =>
-    otherWcifRound !== wcifRound
-    && otherWcifRound.timeLimit
-    && otherWcifRound.timeLimit.cumulativeRoundIds.includes(wcifRound.id)
+export function removeRoundsFromSharedTimeLimits(wcifEvents, wcifRounds) {
+  _.compact(_.flatMap(wcifEvents, 'rounds')).forEach(otherWcifRound =>
+    _.pull(otherWcifRound.timeLimit.cumulativeRoundIds, ...wcifRounds)
```

Comment:
Nitpick: As opposed to the previous `findRoundsSharingTimeLimitWithRound` function, you are now implicitly modifying the `wcifEvents` parameter in-place.

Please document that fact very clearly. If it's not too much of a hassle, try thinking of a way that the exported function doesn't `_.pull` by itself, but rather returns the events that have to be pulled, so that every React component can modify its own state/props based on the return value of this method.

Reply:
I see what you mean. But the components using this helper already have the events that need to be pulled; they're passed in as the parameter `wcifRounds`. Doing what you suggest would basically amount to copy-pasting this helper function into any other component which uses it, unless I'm missing something.
I'm guessing it would be better to add a comment to the current state of the code, as opposed to having two copies of the same function in different files with some comments explaining why?

Reply:
If the alternative would really just amount to copy-pasting, then adding a comment to this method and leaving as-is will be fine.

Reply:
Added comments here and with the import.

Reply:
Thanks!
(a) Is there something like "documentation comments" in JavaScript? For example, in Java `// this is a general comment` and `/** This is a documentation comment */`.
(b) Please refrain from adding comments in the imports section, nobody ever reads these. Instead, add a comment at the call-site if necessary. I think changing the method comment to a proper documentation comment as per (a) will suffice.

Reply:
(a) Yes, seems like it, see [wikipedia](https://en.wikipedia.org/wiki/JSDoc).
(b) Ok makes sense. Something like this?
```
/**
 * Finds the cumulativeRoundIds of each event in wcifEvents and removes any
 * which are found in wcifRounds. Note that it modifies wicfEvents in place.
 *
 * @param {collection} wcifEvents Will be modified in place.
 * @param {Array}      wcifRounds Rounds to be removed from all cumulativeRoundIds.
 */
 ```

Reply:
Yup, something along those lines :)

Reply:
Done. Thanks for bringing comments vs documentation to my attention.